pkg/controller/serverside/reconcile.go

```diff
@@ -39,3 +39,13 @@ func ReconcileObject(ctx context.Context, c client.Client, obj client.Object) er
 
 	return nil
 }
+
+// UpdateObject updates the existing object during reconciliation.
+// This is intended for special use cases only as the preferred method to reconcile objects is server-side apply.
```

Comment:
Given this is a special use-case, shouldn't this live in the package with the special use-case?

Reply:
Hmm yeah good point. I put in this file because it serves a similar purpose as the other methods in this file. So even those we only use this method for this one use-case, there isn't anything specific to reconciling Cilium in this logic. Leaving it here would also allow us to use this same method if it's ever needed in other components.